File: kora-app-annotation-processor/src/main/java/ru/tinkoff/kora/kora/app/annotation/processor/declaration/ModuleDeclaration.java
package ru.tinkoff.kora.kora.app.annotation.processor.declaration;

import javax.lang.model.element.TypeElement;

public interface ModuleDeclaration {
    TypeElement element();

    record MixedInModule(TypeElement element) implements ModuleDeclaration {}

    record AnnotatedModule(TypeElement element) implements ModuleDeclaration {}
}

File: kora-app-symbol-processor/src/main/kotlin/ru/tinkoff/kora/kora/app/ksp/declaration/ModuleDeclaration.kt
package ru.tinkoff.kora.kora.app.ksp.declaration

import com.google.devtools.ksp.symbol.KSClassDeclaration

sealed interface ModuleDeclaration {
    val element: KSClassDeclaration

    data class MixedInModule(override val element: KSClassDeclaration) : ModuleDeclaration
    data class AnnotatedModule(override val element: KSClassDeclaration) : ModuleDeclaration
}
